memory/allocator.go

```diff
@@ -248,19 +252,25 @@ func (a *ResourceAllocator) allocator() memory.Allocator {
 }
 
 type GcAllocator struct {
-	*ResourceAllocator
+	mem *ResourceAllocator
```

Comment:
This should just be `memory.Allocator`. The location that's interested in `MaxAllocated()` and `TotalAllocated()` should keep their reference to their ResourceAllocator. `GcAllocator` doesn't need to know this.

Reply:
No, I need a direct reference to `ResourceAllocator` so I can call `Allocated` etc in tests to verify that we actually free the allocations.

Reply:
You can hold onto both the `ResourceAllocator` and the `GcAllocator` to do that. There's also the `CheckedAllocator` included in arrow for this purpose.

```go
mem := &memory.ResourceAllocator{}
gc := memory.NewGcAllocator(mem)
```

Then you can check `mem` in the tests.